File: Leetcode/2094.go
package Leetcode

import "sort"

func findEvenNumbers(digits []int) []int {
	candidates := make(map[int]int)
	for i, d1 := range digits {
		if d1 == 0 {
			continue
		}
		for j, d2 := range digits {
			if i == j {
				continue
			}
			for k, d3 := range digits {
				if k == i || k == j {
					continue
				}
				if d3&1 != 0 {
					continue
				}
				candidates[d1*100+d2*10+d3] = 1
			}
		}
	}
	ans := make([]int, 0)
	for k, _ := range candidates {
		ans = append(ans, k)
	}
	sort.Ints(ans)
	return ans
}

File: Leetcode/2094.py
from typing import List


class Solution:
    def findEvenNumbers(self, digits: List[int]) -> List[int]:
        arr = digits
        n = len(arr)
        candidates: set[int] = set()
        for i in range(n):
            if arr[i] == 0:
                continue
            for j in range(n):
                if i == j:
                    continue
                for k in range(n):
                    if k == i or k == j:
                        continue
                    if arr[k] & 1:
                        continue
                    candidates.add(arr[i] * 100 + arr[j] * 10 + arr[k])
        return sorted(candidates)
